src/Illuminate/Queue/BeanstalkdQueue.php

```diff
@@ -148,7 +148,7 @@ public function deleteMessage($queue, $id)
      */
     public function getQueue($queue)
     {
-        return $queue ?: $this->default;
+        return $this->getQueuePrefix().($queue ?: $this->default);
```

Comment:
if I'm following you correctly, the problem with moving the `getQueue` method to the parent is each driver has a different implementation.  take a look at the `SqsQueue` file, as it's the weirdest of them all.